File: app/src/main/java/com/example/healthtracker/feature/heartRate/HeartRate.java
package com.example.healthtracker.feature.heartRate;

import com.example.healthtracker.common.base.BasePresenter;
import com.example.healthtracker.common.base.BaseView;
import com.example.healthtracker.model.CalcDao;

import kotlin.Pair;

public interface HeartRate {

    interface View extends BaseView {
        void onRegisterHeartRate();
    }

    interface Presenter extends BasePresenter {
        void registerHeartRateValue(double bpm, String hrClassification, CalcDao dao);
        boolean validate(String age, String bpm);
        Pair<Integer, Pair<Integer, Integer>> getClassificationHeartRate(boolean isMan, int bpm, int age);
    }

}

File: app/src/main/java/com/example/healthtracker/feature/heartRate/HeartRate.kt
package com.example.healthtracker.feature.heartRate

import com.example.healthtracker.common.base.BasePresenter
import com.example.healthtracker.common.base.BaseView
import com.example.healthtracker.model.CalcDao

interface HeartRate {

    interface View: BaseView<Presenter> {
        fun onHeartRateRegister()
    }

    interface Presenter: BasePresenter {
        fun registerHeartRateValue(bpm: Double, hrClassification: String, dao: CalcDao)
        fun validate(age: String, bpm: String): Boolean
        fun getClassificationHeartRate(
            isMan: Boolean, bpm: Int, age: Int
        ): Pair<Int, Pair<Int, Int>>
    }

}